src/huggingface_hub/utils/_errors.py

```diff
@@ -52,27 +52,44 @@ def _raise_for_status(request):
     Internal version of `request.raise_for_status()` that will refine a
     potential HTTPError.
     """
+    request_id = request.headers.get("X-Request-Id")
+
     if "X-Error-Code" in request.headers:
         error_code = request.headers["X-Error-Code"]
         if error_code == "RepoNotFound":
             raise RepositoryNotFoundError(
-                f"404 Client Error: Repository Not Found for url: {request.url}. "
-                "If the repo is private, make sure you are authenticated."
+                f"404 Client Error: Repository Not Found for url: {request.url}. If the"
+                " repo is private, make sure you are authenticated. (Request ID:"
+                f" f{request_id})"
             )
         elif error_code == "RevisionNotFound":
             raise RevisionNotFoundError(
-                f"404 Client Error: Revision Not Found for url: {request.url}"
+                f"404 Client Error: Revision Not Found for url: {request.url}. (Request"
+                f" ID: f{request_id})"
             )
         elif error_code == "EntryNotFound":
             raise EntryNotFoundError(
-                f"404 Client Error: Entry Not Found for url: {request.url}"
+                f"404 Client Error: Entry Not Found for url: {request.url}. (Request"
+                f" ID: f{request_id})"
             )
 
     if request.status_code == 401:
         # The repo was not found and the user is not Authenticated
         raise RepositoryNotFoundError(
-            f"401 Client Error: Repository Not Found for url: {request.url}. "
-            "If the repo is private, make sure you are authenticated."
+            f"401 Client Error: Repository Not Found for url: {request.url}. If the"
+            " repo is private, make sure you are authenticated. (Request ID:"
+            " f{request_id})"
         )
 
-    request.raise_for_status()
+    _raise_with_request_id(request)
+
+
+def _raise_with_request_id(request):
+    request_id = request.headers.get("X-Request-Id")
+    try:
+        request.raise_for_status()
+    except Exception as e:
+        if request_id is not None and len(e.args) and isinstance(e.args[0], str):
```

Comment:
```suggestion
        if request_id is not None and len(e.args) > 0 and isinstance(e.args[0], str):
```
Nit, I usually prefer explicit tests than relying on Python magic conversion to bool.